src/main/java/com/oath/halodb/HaloDBInternal.java

```diff
@@ -424,6 +435,8 @@ private void buildInMemoryIndex(HaloDBOptions options) throws IOException {
 
                 RecordMetaDataForCache existing = inMemoryIndex.get(key);
 
+                maxSequenceNumber = Long.max(sequenceNumber, maxSequenceNumber);
```

Comment:
minor change: could you please move this to the line right after 
`long sequenceNumber = entry.getSequenceNumber();`

Reply:
done